### DISCUSSION TITLE
What causes "Sourcemap for [...] points to missing source files"?

### DISCUSSION THREAD

**Discussion by jtlapp** [sveltejs/kit#4551] [Q&A]
I've spent the day trying to convert my sveltekit repo to a monorepo. After copying my former repo to a package, `yarn run dev --` gives me the following errors:

```
Sourcemap for "/Users/joe/repos/cavesite-new/packages/frontend/src/common/VariableFlash.svelte" points to missing source files
Sourcemap for "/Users/joe/repos/cavesite-new/packages/frontend/src/components/HeaderBar.svelte" points to missing source files
Sourcemap for "/Users/joe/repos/cavesite-new/packages/frontend/src/common/Notice.svelte" points to missing source files
Sourcemap for "/Users/joe/repos/cavesite-new/packages/frontend/src/dialogs/LoginDialog.svelte" points to missing source files
Sourcemap for "/Users/joe/repos/cavesite-new/packages/frontend/src/components/RouteTab.svelte" points to missing source files
```

However, if I place `"mapRoot": "whatever"` into `tsconfig.json`, then I get vastly more such errors, so it's clear that it's happy with many source maps despite the above errors.

I'm using `adapter-node`, and here's my `tsconfig.json`:

```json
{
    "extends": "./.svelte-kit/tsconfig.json",
    "compilerOptions": {
        "strict": true,
        "pretty": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noImplicitReturns": true,
        "noFallthroughCasesInSwitch": true,
        "noEmitOnError": true
    }
}
```

What might cause this problem?

**Comment by jtlapp** [comment]
I don't get these errors in my original repo. I installed svelte-kit from scratch into the new monorepo package and selectively copied files from the old repo in, but I still get the problem.

So either copying svelte config files to a new directory can break things, or it's copying to a pnpm package directory that breaks things.

**Reply by jtlapp** [reply]
I've tried these two config files and everything in between:

```js
import adapter from '@sveltejs/adapter-node';
import preprocess from 'svelte-preprocess';

/** @type {import('@sveltejs/kit').Config} */
const config = {
	preprocess: preprocess(),
	kit: {
		adapter: adapter()
	}
};

export default config;
```

```js
import adapter from '@sveltejs/adapter-node';
import preprocess from 'svelte-preprocess';

/** @type {import('@sveltejs/kit').Config} */
const config = {
	tsconfigFile: "./tsconfig.json",
	compilerOptions: {
		enableSourcemap: true,
	},
	preprocess: preprocess({
		sourceMap: true
	}),
	kit: {
		adapter: adapter()
	}
};

export default config;
```

**Comment by possan** [comment]
I had similar problems with the build step finding files within the node_modules folder, downgrading `vite` seems to have solved it for me, i currently have `"vite": "2.7.1"` in my package.json, not sure if it helps.

**Reply by jtlapp** [reply]
Thank you @possan. Googling around also offered that explanation, but `vite` is a svelte-kit dependency, so I'm not sure how to downgrade it except by downgrading svelte-kit.

**Reply by possan** [reply]
Ah yes, we are not yet on the latest sveltekit and those newer versions might require a more recent vite, i haven't tried that yet.

**Reply by possan** [reply]
Tried it, and yes, 💥

**Reply by jtlapp** [reply]
Wait. You tried to upgrade sveltekit and now you're getting these errors? I was going to try downgrading to see if that helped.

**Comment by bfanger** [comment]
Some (maybe all) are related to postcss generating wrong paths into sourcemap because svelte-preprocess is not passing a `to` parameter.
Fixed in  https://github.com/sveltejs/svelte-preprocess/pull/500

**Reply by jtlapp** [reply]
That could be. But in my case, the code was not producing these errors until I split it into separate packages of a monorepo.

**Reply by bfanger** [reply]
Edit both `packages/frontend/node_modules/svelte-preprocess/dist/transformers/globalStyle.js` & `postcss.js` 

and replace:
 ```
from: filename,
```
 with
 ```
from: filename, 
to: filename,
```

In my tests with a monorepo and the example SvelteKit app this fixed the issue.

**Reply by benmccann** [reply]
`svelte-preprocess` 4.10.6 has been released. Please upgrade and let us know if it works. If it does not then please provide a repository that reproduces the issue

**Reply by jtlapp** [reply]
Thanks for producing a fix! Unfortunately, I didn't keep my very broken pnpm monorepo, so I can't test this. But it sounds like @possan also faced this issue attempting to upgrade, so maybe he can check it out.

**Reply by djavorek** [reply]
@benmccann Thank you. I have faced this issues, looks OK now with: ` "svelte-preprocess": "4.10.6",`
